app/lib/shared/handler_helpers.dart

```diff
@@ -125,15 +131,56 @@ shelf.Handler _logRequestWrapper(Logger logger, shelf.Handler handler) {
     }
     try {
       return await handler(request);
-    } catch (error, st) {
-      logger.severe('Request handler failed', error, Trace.from(st));
+    } finally {
+      if (shouldLog) {
+        logger.info('Request handler done.');
+      }
+    }
+  };
+}
 
-      final title = 'Pub is not feeling well';
-      Map<String, String> debugHeaders;
-      if (context.traceId != null) {
-        debugHeaders = {'package-site-request-id': context.traceId};
+shelf.Handler _exceptionHandlerWrapper(Logger logger, shelf.Handler handler) {
+  return (shelf.Request request) async {
+    if (requestContext.emitsJsonResponse) {
```

Comment:
We could also decide this based on the `accept` header... I'm not suggesting we do :)

Reply:
Does `pub` client emits that header? We can move our client-side codes to emit it, and that way it would be much simpler. I'd think that in such case we could omit the `requestContext` part entirely...

Reply:
PTAL, I've updated the code to use the `Accept:` header.